lib/src/react16_suggestors/react_style_maps_updater.dart

```diff
@@ -0,0 +1,299 @@
+// Copyright 2019 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:analyzer/analyzer.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:codemod/codemod.dart';
+import 'package:over_react_codemod/src/react16_suggestors/constants.dart';
+import 'package:source_span/source_span.dart';
+
+/// Suggestor that updates to React 16's StyleMap standard.
+///
+/// Specifically, this suggestor looks for instances where a simple unitless
+/// string literal is passed to a CSS property. React 16 specifies that such
+/// cases should be a num instead of a string.
+class ReactStyleMapsUpdater extends GeneralizingAstVisitor
+    with AstVisitingSuggestorMixin
+    implements Suggestor {
+  @override
+  visitCascadeExpression(CascadeExpression node) {
+    super.visitCascadeExpression(node);
+
+    for (Expression cascade in node.cascadeSections) {
+      if (!hasValidationComment(node, sourceFile)) {
+        if (cascade.toString().contains('style') &&
+            !cascade.toString().contains('setProperty')) {
+          /// A style map, method invocation, or a variable
+          dynamic stylesObject = getStyles(cascade);
+
+          /// CSS properties that were modified by the script or need to be
+          /// manually checked
+          List<String> affectedValues = [];
+
+          // Variables that affect the writing of the comment that goes above
+          // the style prop. There are different messages patched based on
+          // these booleans.
+          bool styleMapContainsAVariable = false;
+          bool isAVariable = false;
+          bool isAFunction = false;
+          bool isForCustomProps =
+              cascade.toString().toLowerCase().contains('props');
+
+          if (stylesObject is MapLiteral) {
+            stylesObject.entries.forEach((MapLiteralEntry cssPropertyRow) {
+              String originalCssPropertyKey =
+                  cssPropertyRow.beginToken.toString();
```

Comment:
Is there a reason not to use `cssPropertyRow.key`/`.value`? That would be more robust than traversing the tokens

Reply:
Nope, no reason! Good call!